src/transmission_electron.py

```diff
@@ -0,0 +1,274 @@
+"""Wrapper for the TEM Simulator."""
+import yaml
+
+
+class TEMSimulator:
+    """Wrapper for the TEM Simulator.
+
+    Parameters
+    ----------
+    path_config : str
+        Relative path to YAML file containing file paths for TEM Simulator.
+    sim_config : str
+        Relative path to YAML file containing simulator paths for TEM Simulator.
+
+    """
+
+    def __init__(self, path_config, sim_config):
+        self.file_paths = self._getConfigFromYaml(path_config)
+        self.sim_dict = self._getConfigFromYaml(sim_config)
+        self.placeholder = 0
+
+    def run_sim(self, pdb_file):
+        """Run TEM simulator on input file and produces particle stacks with metadata.
+
+        Parameters
+        ----------
+        pdb_file : str
+            Relative file path to input .pdb file for sim
+
+        Returns
+        -------
+        particles : arr
+            Individual particle data extracted from micrograph
+        """
+        # self.file_paths = self._getIOFilePaths(pdb_file)
+        # self._buildCordFile(**self.file_paths, **self.sim_dict)
+        # self.param_dict = self._build_param_dict(**self.file_paths, **self.sim_dict)
+        # self._build_inpFile(self.param_dict, **self.file_paths)
+        #
+        # data = self._getImageData(**self.filePaths)
+        pdb_file = "placeholder_to_pass_tests"
+        particles = [self.placeholder, pdb_file]
+        return particles
+
+    def get_config_from_yaml(self, config_yaml):
+        """Create dictionary with parameters from YAML file and groups them into lists.
+
+        Parameters
+        ----------
+        config_yaml : str
+            Relative path to YAML file containing parameters for TEM Simulator
+        Returns
+        -------
+        classified_params : dict
+            Dictionary containing grouped parameters for TEM Simulator, with keys:
+                seed : str maps to int
+                    Seed for TEM Simulator
+                particle_mrcout : str maps to bool
+                    Flag for optional volume map of sample
+                sample_dimensions : str maps to
```

Comment:
str maps to ... ?

Reply:
should` sample_dimensions` be called `specimen_grid_params`? it would be more explanatory and consistent with the other namings

Reply:
`specimen_grid_params` is also the name that appears in `sim_config.yml`.